x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/installation/types.ts

```diff
@@ -57,13 +57,19 @@ export interface EntityEngineInstallationDescriptor {
 
   /**
    * The ingest pipeline to apply to the entity data.
-   * This can be an array of processors which get appended to the default pipeline,
-   * or a function that takes the default processors and returns an array of processors.
+   * This can be an array of processors which get appended to the platform pipeline,
+   * or a function that takes the platform processors and returns an array of processors.
    **/
   pipeline?:
     | IngestProcessorContainer[]
     | ((defaultProcessors: IngestProcessorContainer[]) => IngestProcessorContainer[]);
 
+  /**
+   * Custom processors to apply to the entity data.
+   * It's an array of processors which get appended to the platform pipeline of the defined entity.
```

Comment:
I wonder if we need this new customIngestProcessors or just make `pipelines` do what it claims it does. 

In theory we could export the `pipeline` [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_definitions/types.ts#L16-L28) and define it in the definitions.

Then on this [convoluted logic](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/elasticsearch_assets/ingest_pipeline.ts#L117-L119), append the processors of the pipeline case there are any (right now we support only function for some reason)

Reply:
Really good point @romulets, I was initially confused about how the `pipeline` parameter should work, but after reviewing it you were right, the `pipeline` parameter should already handle that however, it was working only with function.

I updated it to make it work as intended, so we can override it with either of the following:

```tsx
  pipeline: (defaultProcessors) => [
    ...defaultProcessors,
    {
      set: {
        field: 'entity.type',
        value: 'Identity',
      },
    },
  ],
```

or 

```tsx
  pipeline: [
    {
      set: {
        field: 'entity.type',
        value: 'Host',
      },
    },
  ],
```